fda/depth_measures.py

```diff
@@ -0,0 +1,388 @@
+                [0.83333333],
+                [0.83333333],
+                [0.83333333],
+                [0.83333333],
+                [0.83333333]],
+        <BLANKLINE>
+               [[0.5       ],
+                [0.5       ],
+                [0.83333333],
+                [0.83333333],
+                [0.83333333],
+                [0.83333333]],
+        <BLANKLINE>
+               [[0.83333333],
+                [0.83333333],
+                [0.83333333],
+                [0.5       ],
+                [0.5       ],
+                [0.5       ]]]))
+
+        Multivariate Setting without specifying pointwise:
+
+        >>> data_matrix = [[[[1, 3], [2, 6]], [[23, 54], [43, 76]], [[2, 45], [12, 65]]],
+        ...                [[[21, 34], [8, 16]], [[67, 43], [32, 21]], [[10, 24], [3, 12]]],
+        ...                [[[4, 6], [4, 10]], [[45, 48], [38, 56]], [[34, 78], [10, 28]]]]
+        >>> sample_points = [[2, 4, 6], [3, 6]]
+        >>> fd = FDataGrid(data_matrix, sample_points)
+        >>> modified_band_depth(fd)
+        array([[0.66666667, 0.72222222],
+               [0.72222222, 0.66666667],
+               [0.94444444, 0.94444444]])
+
+    """
+    if fdatagrid.ndim_domain > 2:
+        raise NotImplementedError("Only support 1 or 2 dimensions on the domain.")
+
+    if fdatagrid.ndim_domain == 1:
+        axis = 1
+        npoints_sample = fdatagrid.ncol
+    else:
+        axis = (1, 2)
+        npoints_sample = len(fdatagrid.sample_points[0]) * len(fdatagrid.sample_points[1])
+
+    n = fdatagrid.nsamples
+    nchoose2 = n * (n - 1) / 2
+
+    ranks = _rank_samples(fdatagrid)
+    nsamples_above = fdatagrid.nsamples - ranks
+    nsamples_below = ranks - 1
+    match = nsamples_above * nsamples_below
+    proportion = match.sum(axis=axis) / npoints_sample
+    depth = (proportion + fdatagrid.nsamples - 1) / nchoose2
+
+    if pointwise:
+        depth_pointwise = (match + fdatagrid.nsamples - 1) / nchoose2
+        return depth, depth_pointwise
+    else:
+        return depth
+
+
+def _cumulative_distribution(column):
+    """Calculates the cumulative distribution function of the values passed to the function and evaluates it at each point.
+
+    Args:
+        column (numpy.darray): Array containing the values over which the distribution function is calculated.
+
+    Returns:
+        numpy.darray: Array containing the evaluation at each point of the distribution function.
+
+    Examples:
+        >>> _cumulative_distribution(np.array([1, 4, 5, 1, 2, 2, 4, 1, 1, 3]))
+        array([0.4, 0.9, 1. , 0.4, 0.6, 0.6, 0.9, 0.4, 0.4, 0.7])
+
+    """
+    if len(column.shape) != 1:
+        raise ValueError("Only supported 1 dimensional arrays.")
+    _, indexes, counts = np.unique(column, return_inverse=True, return_counts=True)
+    count_cumulative = np.cumsum(counts) / len(column)
+    return count_cumulative[indexes].reshape(column.shape)
+
+
+def Fraiman_Muniz_depth(fdatagrid, pointwise=False):
```

Comment:
Functions should be in lowercase.

Reply:
But is a proper name

Reply:
A lot of distances [here](https://docs.scipy.org/doc/scipy/reference/spatial.distance.html) also are proper names, but are still in lowercase.

Reply:
changed